storage/src/nodestore/persist.rs

```diff
@@ -199,30 +279,30 @@ impl<S: WritableStorage + 'static> NodeStore<Committed, S> {
     /// Returns a [`FileIoError`] if any node cannot be written to storage.
     #[fastrace::trace(short_name = true)]
     pub fn flush_nodes(&mut self) -> Result<NodeStoreHeader, FileIoError> {
-        #[cfg(feature = "io-uring")]
-        if let Some(this) = self.downcast_to_file_backed() {
-            return this.flush_nodes_io_uring();
-        }
-        self.flush_nodes_generic()
-    }
+        let flush_start = Instant::now();
+
+        let header = {
+            #[cfg(feature = "io-uring")]
+            {
+                use crate::FileBacked;
+                // Try to use io-uring if available and storage type supports it
+                let this = self as &mut dyn std::any::Any;
+                if let Some(file_backed) = this.downcast_mut::<NodeStore<Committed, FileBacked>>() {
+                    file_backed.flush_nodes_io_uring()?
+                } else {
+                    self.flush_nodes_generic()?
+                }
+            }
+            #[cfg(not(feature = "io-uring"))]
+            {
+                self.flush_nodes_generic()?
+            }
+        };
```

Comment:
I would probably leave the `downcast_to_file_backed` method as this removes the documentation as to why it exists and looks more unwieldily. But I think I'm forming an idea of a better way of doing this that doesn't require this `dyn Any` downcast so maybe not worth changing.